pkg/client/multi.go

```diff
@@ -68,15 +84,10 @@ func isNil(obj kclient.Object) bool {
 
 func onOneList[T any, V ObjectPointer[T]](ctx context.Context, factory ProjectClientFactory, name string, cb func(name string, client Client) ([]T, error)) ([]T, error) {
 	var (
-		result      []T
-		projectName = ""
+		result []T
 	)
-	i := strings.LastIndex(name, "/")
-	if i != -1 {
-		projectName = name[0:i]
-		name = name[i+1:]
-	}
-	client, err := factory.ForProject(ctx, projectName)
+	// parse the form project::resource
+	client, name, err := parseProject(ctx, name, factory)
```

Comment:
The `ForFoo` probably doesn't make as much sense here since the `factory` prefix is no longer used. If I had to make a call here, I'd pick something along the lines of `resourceClient`.

Reply:
Alternatively, you could add the `parseProject` function into the `factory` package and change the name to `ForResource`.

Reply:
Don't want to move it under factory, as I want it to remain a function that accepts the factory interface. This was carefully implemented to not break mocks. Renamed to getResourceAndClient